### PR TITLE
Add specialised lanes for app beta and release deployment

### PR DESCRIPTION
**Pull request by kairadiagne**
Adds several lanes related to app deployment and provisioning.

### CODE REVIEW

**Comment by wetransferplatform** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>Please provide a summary in the Pull Request description</td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  View more details on <a href="https://app.bitrise.io/build/7c8a5be5eecca13d" target="_blank">Bitrise</a>
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>WeTransferPRLinter-Package: Executed 26 tests, with 0 failures (0 unexpected) in 1.944 (2.004) seconds</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 34f72da17f7f5968e95631561cf69713c90d13f3
</p>

File: Fastlane/Fastfile [hunk 1]
```diff
@@ -87,92 +98,74 @@ lane :test_project do |options|
...
-	# Create a PR
-    # Create a pull request for master to include the updated Changelog.md and podspec
-    create_pull_request(
-      api_token: ENV["DANGER_GITHUB_API_TOKEN"],
-      repo: git_repository_name,
-      title: "Merge release #{latest_tag} into master",
-      base: "master", # The branch to merge the changes into.
-      body: "Containing all the changes for our [**#{latest_tag} Release**](#{release_url})."
-    )
+  sh("git push origin release/#{latest_tag}")
```
Comment (AvdLee) [inline Fastlane/Fastfile:142]: It looks like indenting is a bit off on some places

File: Fastlane/deployment_lanes.rb [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+
+  increment_build_number(build_number: new_build_number)
+end
+
+desc "Syncs production certificates on CI"
+desc ""
+desc "#### Options"
+desc " * **`app_identifier`**: a list of all app identifiers for which to sync the certs"
+desc ""
+private_lane :certs do |options|
```
Comment (AvdLee) [inline Fastlane/deployment_lanes.rb:345]: Shouldn't this be part of the provisioning lanes file?

File: Gemfile [hunk 3]
```diff
@@ -6,3 +6,4 @@ gem 'fastlane'
 gem 'xcpretty'
 gem 'xcpretty-json-formatter'
 gem 'cocoapods'
+gem 'dotenv'
```
Comment (AvdLee) [inline Gemfile:9]: Learned something new!

**Review by AvdLee — CHANGES_REQUESTED** [review]
Really well done, especially on the documentation! I can see that becoming more valuable now that the team grows 💪

File: Fastlane/Fastfile (hunk 1 above)
Comment (kairadiagne) [reply Fastlane/Fastfile:143]: Yes it is not aligned anymore. It happened in more places, therefore I used rubicon to lint all the files.

File: Fastlane/deployment_lanes.rb (hunk 2 above)
Comment (kairadiagne) [reply Fastlane/deployment_lanes.rb:348]: I decided to make it a private lane as it should only be used on CI. The provisioning lanes can be used locally on the command line. Let me know if that makes sense.

Comment (AvdLee) [reply Fastlane/deployment_lanes.rb:348]: wasn't aware of private lanes, awesome!

**Review by AvdLee — APPROVED** [review]

**Pull request merged**